tests/Feature/RoomTest.php

```diff
@@ -45,4 +45,50 @@ public function test_getAll_rooms_success(): void
             });
 
     }
+
+    public function test_getSingleRoomValid(): void
+    {
+        $room = Room::factory()->create();
+
+        $response = $this->getJson('/api/rooms/1');
+
+        $response->assertOk()
+            ->assertJson(function (AssertableJson $json) {
+                $json->hasAll(['message', 'data'])
+                    ->has('data', function (AssertableJson $json) {
+                        $json->hasAll(['id', 'name', 'rate', 'image', 'min_capacity', 'max_capacity', 'description', 'type'])
+                            ->whereAllType([
+                                'id' => 'integer',
+                                'name' => 'string',
+                                'rate' => 'integer',
+                                'image' => 'string',
+                                'min_capacity' => 'integer',
+                                'max_capacity' => 'integer',
+                                'description' => 'string',
+                            ])
+                            ->has('type', function (AssertableJson $json) {
+                                $json->hasAll(['id', 'name'])
+                                    ->whereAllType([
+                                        'id' => 'integer',
+                                        'name' => 'string',
+                                    ]);
+                            }
+
+                            );
+                    });
+            });
+    }
+
+    public function test_getSingleRoomInvalid(): void
+    {
+        Room::factory()->create();
```

Comment:
No need to create a room here, just because you don't actually use it in your test.